src/mbgl/map/map.cpp

```diff
@@ -825,15 +829,34 @@ std::unique_ptr<Source> Map::removeSource(const std::string& sourceID) {
     }
     return nullptr;
 }
+    
+std::vector<style::Layer*> Map::getLayers() {
+    return impl->style ? impl->style->getLayers() : std::vector<style::Layer*>();
+}
 
-style::Layer* Map::getLayer(const std::string& layerID) {
+Layer* Map::getLayer(const std::string& layerID) {
     if (impl->style) {
         impl->styleMutated = true;
         return impl->style->getLayer(layerID);
     }
     return nullptr;
 }
 
+void Map::setLayers(std::vector<std::unique_ptr<Layer>> layers) {
+    if (!impl->style) {
+        impl->styleMutated = true;
```

Comment:
👍 I ended up not using this method in the SDK due to the memory management conundrum in the PR description. Should I keep this method around to complement `getLayers()` or remove it?